runtime/doc/lua.txt

```diff
@@ -3949,6 +3949,205 @@ vim.net.request({url}, {opts}, {on_response})              *vim.net.request()*
                        success.
 
 
+==============================================================================
+Lua module: vim.pos                                                  *vim.pos*
+
+WARNING: This module is under experimental support. Its semantics are not yet
+finalized, and the stability of this API is not guaranteed. Avoid using it
+outside of Nvim. You may subscribe to or participate in the tracking issue
+https://github.com/neovim/neovim/issues/25509 to stay updated or contribute to
+its development.
+
+Built on |vim.Pos| objects, this module offers operations that support
+comparisons and conversions between various types of positions.
+
+
+*vim.Pos*
+    Represents a well-defined position.
+
+    A |vim.Pos| object contains the {row} and {col} coordinates of a position.
+    To create a new |vim.Pos| object, call `vim.pos()`.
+
+    Example: >lua
+        local pos1 = vim.pos(3, 5)
+        local pos2 = vim.pos(4, 0)
+
+        -- Operators are overloaded for comparing two `vim.Pos` objects.
+        if pos1 < pos2 then
+          print("pos1 comes before pos2")
+        end
+
+        if pos1 ~= pos2 then
+          print("pos1 and pos2 are different positions")
+        end
+<
+
+    It may include optional fields that enable additional capabilities, such
+    as format conversions.
+
+    Fields: ~
+      • {row}     (`integer`) 0-based byte index.
+      • {col}     (`integer`) 0-based byte index.
+      • {buf}?    (`integer`) Optional buffer handle.
+
+                  When specified, it indicates that this position belongs to a
+                  specific buffer. This field is required when performing
+                  position conversions.
+      • {to_lsp}  (`fun(pos: vim.Pos, position_encoding: lsp.PositionEncodingKind)`)
+                  See |Pos:to_lsp()|.
+      • {lsp}     (`fun(buf: integer, pos: lsp.Position, position_encoding: lsp.PositionEncodingKind)`)
+                  See |Pos:lsp()|.
+
+
+Pos:lsp({buf}, {pos}, {position_encoding})                         *Pos:lsp()*
+    Creates a new |vim.Pos| from `lsp.Position`.
```

Comment:
Does this make sense as a method (`pos:lsp()`) or should it be strictly a module function (`vim.pos.lsp()`)?